### ISSUE TITLE
`prefer-single-boolean-return` clarification

### ISSUE THREAD

**Issue by zloirock** [SonarSource/eslint-plugin-sonarjs#329]
After #324 this code this code reports `sonarjs/prefer-single-boolean-return`:

```js
function isTemplateObject(value) {
  if (!isFrozenStringArray(value, true)) return false;
  var raw = value.raw;
  if (raw.length !== value.length || !isFrozenStringArray(raw, false)) return false;
  return true;
}
```

How should I rewrite it (getting `value.raw` is an observable operation)?

**Comment by vilchik-elena** [comment]
I am not sure what is observable operation, but how about this?
```js
function isTemplateObject(value) {
  if (!isFrozenStringArray(value, true)) return false;
  var raw = value.raw;
  return raw.length === value.length && isFrozenStringArray(raw, false);
}
```

**Comment by B2o5T** [comment]
```diff
- if (raw.length !== value.length || !isFrozenStringArray(raw, false)) return false;
-  return true;
+ return raw.length === value.length || isFrozenStringArray(raw, false)
```

**Comment by zloirock** [comment]
Ah, thanks, I thought that the first `if` is also taken into account and recommended something like
```js
function isTemplateObject(value) {
  var raw;
  return isFrozenStringArray(value, true)
    && (raw = value.raw).length === value.length
    && isFrozenStringArray(raw, false);
}
```

**Comment by B2o5T** [comment]
@zloirock Keep attention to that suggestion from @vilchik-elena containing different behaviour of your function!

**Comment by zloirock** [comment]
@B2o5T here case is correct since the condition is inverted.
